bootkube/resources/charts/calico/templates/host-endpoint-controller.yaml

```diff
@@ -70,3 +71,4 @@ roleRef:
   kind: ClusterRole
   name: calico-hostendpoint-controller-role
   apiGroup: rbac.authorization.k8s.io
+{{- end }}
```

Comment:
The controller only works in packet (it has "bond0" hardcoded, etc.). Any reason to move it to the control plane? Just to have it in a general place?

The advantage that I see is that, as it is right now, uses `kubectl` in a simple script and we should keep it in sync with kubernetes version. But not sure about your reasoning and curious about it :)

Reply:
Well, the advantage is, that we maintain YAMLs in a single place. For now it will only be used in Packet, but eventually, I think this controller should be added to Calico Kubernetes controller, so it will still be part of this chart.

That also makes updates easier, as all manifests are handled in the same ways, so we can skip resources in Terraform.

>The advantage that I see is that, as it is right now, uses kubectl in a simple script and we should keep it in sync with kubernetes version. But not sure about your reasoning and curious about it :)

How to we ensure this sync right now?

Reply:
> Well, the advantage is, that we maintain YAMLs in a single place. For now it will only be used in Packet, but eventually, I think this controller should be added to Calico Kubernetes controller, so it will still be part of this chart.

Why? All other platforms (maybe not baremetal, depends on the setup) have firewalls by the cloud provider (security group in aws, etc.). People usually use that, even for kubernetes setups. Why would they use this?

> How to we ensure this sync right now?

We don't have a process, we need to remember. I think it was a follow-up for some issue to see how to do this.

Reply:
The only rationale for introducing GNPs in the first place was lack of provider-level firewalling on Packet. On AWS and similar platforms, using security groups is significantly safer than using GNPs, for multiple reasons. Therefore, I don't see why we would want GNPs on all platforms.